src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/config/OpenApiConfiguration.kt

```diff
@@ -0,0 +1,128 @@
+
+import io.swagger.v3.core.util.PrimitiveType
+import io.swagger.v3.oas.models.Components
+import io.swagger.v3.oas.models.OpenAPI
+import io.swagger.v3.oas.models.Operation
+import io.swagger.v3.oas.models.info.Contact
+import io.swagger.v3.oas.models.info.Info
+import io.swagger.v3.oas.models.media.DateTimeSchema
+import io.swagger.v3.oas.models.media.Schema
+import io.swagger.v3.oas.models.media.StringSchema
+import io.swagger.v3.oas.models.security.SecurityRequirement
+import io.swagger.v3.oas.models.security.SecurityScheme
+import io.swagger.v3.oas.models.servers.Server
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springdoc.core.customizers.OpenApiCustomizer
+import org.springdoc.core.customizers.OperationCustomizer
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.info.BuildProperties
+import org.springframework.context.ApplicationContext
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.expression.BeanFactoryResolver
+import org.springframework.expression.spel.SpelEvaluationException
+import org.springframework.expression.spel.standard.SpelExpressionParser
+import org.springframework.expression.spel.support.StandardEvaluationContext
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.method.HandlerMethod
+
+@Configuration
+class OpenApiConfiguration(buildProperties: BuildProperties) {
+  private val version: String = buildProperties.version
+
+  @Autowired
+  private lateinit var context: ApplicationContext
+
+  @Bean
+  fun customOpenAPI(): OpenAPI {
+    return OpenAPI().servers(
+      listOf(
+        Server().url("https://assess-for-early-release-api-dev.hmpps.service.justice.gov.uk")
+          .description("Development"),
+        Server().url("http://localhost:8089").description("Local"),
+      ),
+    ).info(
+      Info().title("Assess for early release API").version(version)
+        .description("API for the Assess for early release product. NB: Not intended for general use")
+        .contact(Contact().name("HMPPS Digital Studio").email("feedback@digital.justice.gov.uk")),
+    ).components(
+      Components().addSecuritySchemes(
+        "assess-for-early-release-admin-role",
+        SecurityScheme().addBearerJwtRequirement("ROLE_ASSESS_FOR_EARLY_RELEASE_ADMIN"),
+      ),
+    ).addSecurityItem(SecurityRequirement().addList("assess-for-early-release-admin-role", listOf("read", "write")))
+  }
+
+  private fun SecurityScheme.addBearerJwtRequirement(role: String): SecurityScheme =
+    type(SecurityScheme.Type.HTTP).scheme("bearer").bearerFormat("JWT").`in`(SecurityScheme.In.HEADER)
+      .name("Authorization").description("A HMPPS Auth access token with the `$role` role.")
+
+  @Bean
+  fun preAuthorizeCustomizer(): OperationCustomizer {
```

Comment:
This reads the PreAuthorize annotation to extract out the role required and add it to the swagger doc